sdk/core/azure-core/src/main/java/com/azure/android/core/http/CallbackWithHeader.java

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.core.http;
+
+/**
+ * Callback to receive a service operation result.
+ *
+ * @param <T> The type of the result.
+ * @param <T> The type of the header.
+ */
+public interface CallbackWithHeader<T, H> {
+    /**
+     * The method to call on a successful result.
+     *
+     * @param result   The result.
+     * @param header   The custom header value.
+     * @param response The response.
+     */
+    void onSuccess(T result, H header, okhttp3.Response response);
```

Comment:
It will be a single type. Basically swagger will define a header class type (H), whose properties are set from subset of raw http response headers. This often means those headers has special meaning or important for the service. All storage apis uses this pattern.

Reply:
But they have to be part of okhttp3.Response.getHeaders() to be transported across, correct? If that's true, it is already contained in Response.getHeaders(). It is just when we interpret these headers, we need some kind of convenience, e.g. getBodyLength() or hasPagingHeader()/getPagingHeader(). To me, these convenience functions more belong to Response<T> class. So instead of CallbackWithHeader, we may consider having Callback<Response<T>> where Response<T> class may interpret the raw okhttp3 headers for our convenience.

Reply:
For example, if ListBlobFlatSegmentHeaders is expected from the response, a class derived from Response\<T\>, e.g. ListBlobResponse, may provide function like getListBlobFlatSegmentHeaders(): ListBlobFlatSegmentHeaders where the corresponding raw header value is parsed into ListBlobFlatSegmentHeaders. When callback.onSuccess(listBlobResponse) is called, the header is already parsed. This way parsing of that particular header value will only happen in one place and one place only, in this case in ListBlobResponse only.

Reply:
To me, reuse existing code, ListBlobResponse, makes more sense. But if you have good reasons, I'll pass then.